File: crates/breez-sdk/cli/src/command/issuer.rs
use breez_sdk_spark::{
    BurnIssuerTokenRequest, CreateIssuerTokenRequest, FreezeIssuerTokenRequest,
    MintIssuerTokenRequest, TokenIssuer, UnfreezeIssuerTokenRequest,
};
use clap::{ArgAction, Subcommand};

use crate::command::print_value;

#[derive(Clone, Debug, Subcommand)]
pub enum IssuerCommand {
    /// Gets the issuer token balance
    TokenBalance,
    /// Gets the issuer token metadata
    TokenMetadata,
    /// Creates a new issuer token
    CreateToken {
        /// Name of the token
        name: String,
        /// Ticker symbol of the token
        ticker: String,
        /// Number of decimal places for the token
        decimals: u32,
        /// Whether the token is freezable
        #[arg(short = 'f', long, action = ArgAction::SetTrue)]
        is_freezable: bool,
        /// Maximum supply of the token
        max_supply: u128,
    },
    /// Mints supply of the issuer token
    MintToken {
        /// Amount of the supply to mint
        amount: u128,
    },
    /// Burns supply of the issuer token
    BurnToken {
        /// Amount of the supply to burn
        amount: u128,
    },
    /// Freezes issuer tokens held at the specified address
    FreezeToken {
        /// Address holding the tokens to freeze
        address: String,
    },
    /// Unfreezes issuer tokens held at the specified address
    UnfreezeToken {
        /// Address holding the tokens to unfreeze
        address: String,
    },
}

pub async fn handle_command(
    token_issuer: &TokenIssuer,
    command: IssuerCommand,
) -> Result<bool, anyhow::Error> {
    match command {
        IssuerCommand::TokenBalance => {
            let response = token_issuer.get_issuer_token_balance().await?;
            print_value(&response)?;
            Ok(true)
        }
        IssuerCommand::TokenMetadata => {
            let metadata = token_issuer.get_issuer_token_metadata().await?;
            print_value(&metadata)?;
            Ok(true)
        }
        IssuerCommand::CreateToken {
            name,
            ticker,
            decimals,
            is_freezable,
            max_supply,
        } => {
            let metadata = token_issuer
                .create_issuer_token(CreateIssuerTokenRequest {
                    name,
                    ticker,
                    decimals,
                    is_freezable,
                    max_supply,
                })
                .await?;
            print_value(&metadata)?;
            Ok(true)
        }
        IssuerCommand::MintToken { amount } => {
            let payment = token_issuer
                .mint_issuer_token(MintIssuerTokenRequest { amount })
                .await?;
            print_value(&payment)?;
            Ok(true)
        }
        IssuerCommand::BurnToken { amount } => {
            let payment = token_issuer
                .burn_issuer_token(BurnIssuerTokenRequest { amount })
                .await?;
            print_value(&payment)?;
            Ok(true)
        }
        IssuerCommand::FreezeToken { address } => {
            let response = token_issuer
                .freeze_issuer_token(FreezeIssuerTokenRequest { address })
                .await?;
            print_value(&response)?;
            Ok(true)
        }
        IssuerCommand::UnfreezeToken { address } => {
            let response = token_issuer
                .unfreeze_issuer_token(UnfreezeIssuerTokenRequest { address })
                .await?;
            print_value(&response)?;
            Ok(true)
        }
    }
}

File: crates/breez-sdk/bindings/examples/cli/langs/wasm/src/issuer.js
'use strict'

const { Command } = require('commander')
const { printValue } = require('./serialization')

/**
 * Register all issuer subcommands on the given commander program.
 *
 * @param {Command} program - The parent commander program (or subcommand)
 * @param {() => object} getTokenIssuer - Function that returns the TokenIssuer instance
 */
function registerIssuerCommands(program, getTokenIssuer) {
  const issuer = program
    .command('issuer')
    .description('Issuer related commands')

  // --- token-balance ---
  issuer
    .command('token-balance')
    .description('Get the issuer token balance')
    .action(async () => {
      const tokenIssuer = getTokenIssuer()
      const response = await tokenIssuer.getIssuerTokenBalance()
      printValue(response)
    })

  // --- token-metadata ---
  issuer
    .command('token-metadata')
    .description('Get the issuer token metadata')
    .action(async () => {
      const tokenIssuer = getTokenIssuer()
      const metadata = await tokenIssuer.getIssuerTokenMetadata()
      printValue(metadata)
    })

  // --- create-token ---
  issuer
    .command('create-token')
    .description('Create a new issuer token')
    .argument('<name>', 'Name of the token')
    .argument('<ticker>', 'Ticker symbol of the token')
    .argument('<decimals>', 'Number of decimal places for the token')
    .argument('<max_supply>', 'Maximum supply of the token')
    .option('-f, --is-freezable', 'Whether the token is freezable', false)
    .action(async (name, ticker, decimals, maxSupply, options) => {
      const tokenIssuer = getTokenIssuer()
      const metadata = await tokenIssuer.createIssuerToken({
        name,
        ticker,
        decimals: parseInt(decimals, 10),
        isFreezable: options.isFreezable,
        maxSupply: BigInt(maxSupply)
      })
      printValue(metadata)
    })

  // --- mint-token ---
  issuer
    .command('mint-token')
    .description('Mint supply of the issuer token')
    .argument('<amount>', 'Amount of the supply to mint')
    .action(async (amount) => {
      const tokenIssuer = getTokenIssuer()
      const payment = await tokenIssuer.mintIssuerToken({
        amount: BigInt(amount)
      })
      printValue(payment)
    })

  // --- burn-token ---
  issuer
    .command('burn-token')
    .description('Burn supply of the issuer token')
    .argument('<amount>', 'Amount of the supply to burn')
    .action(async (amount) => {
      const tokenIssuer = getTokenIssuer()
      const payment = await tokenIssuer.burnIssuerToken({
        amount: BigInt(amount)
      })
      printValue(payment)
    })

  // --- freeze-token ---
  issuer
    .command('freeze-token')
    .description('Freeze issuer tokens held at the specified address')
    .argument('<address>', 'Address holding the tokens to freeze')
    .action(async (address) => {
      const tokenIssuer = getTokenIssuer()
      const response = await tokenIssuer.freezeIssuerToken({
        address
      })
      printValue(response)
    })

  // --- unfreeze-token ---
  issuer
    .command('unfreeze-token')
    .description('Unfreeze issuer tokens held at the specified address')
    .argument('<address>', 'Address holding the tokens to unfreeze')
    .action(async (address) => {
      const tokenIssuer = getTokenIssuer()
      const response = await tokenIssuer.unfreezeIssuerToken({
        address
      })
      printValue(response)
    })
}

module.exports = { registerIssuerCommands }
